src/components/Modal/modal.tsx

```diff
@@ -0,0 +1,175 @@
+import * as React from 'react';
+import { themr, ThemedComponentClass } from 'react-css-themr';
+import { Keys } from '../../types';
+import { MODAL } from '../ThemeIdentifiers';
+import KeypressListener from '../KeypressListener';
+import Helpers from './helpers';
+import Dialog from './dialog';
+import * as baseTheme from './Modal.scss';
+
+const bodyStyle = (pading: string, overflow: string): void => {
+  const body = document.getElementsByTagName('body');
+  body[0].style.paddingRight = pading;
+  body[0].style.overflow = overflow;
+};
+
+const getModalElement = (id: string) => {
+  return {
+    modal: Helpers.getElement(`modal-${id}`),
+    dialog: Helpers.getElement(`dialog-${id}`),
+  };
+};
+
+export type Size = 'Small' | 'Medium' | 'Large' | number;
+
+export interface Ianimate {
+  in: Function,
+  out: Function,
+}
+
+export interface Itrigger {
+  body: string,
+  animate: Ianimate,
+}
+
+export interface Props {
+  className?: string,
+  classes?: string,
+  close?: boolean,
+  footer?: React.ReactNode,
+  header?: string | React.ReactNode,
+  activator: React.ReactElement<any>,
+  id?: string,
+  show: boolean,
+  trigger: Itrigger,
+  closeOnEsc?: boolean,
+  closeOnBackgroud?: boolean,
+  modalOverflow?: boolean,
+  backdropEnabled?: boolean,
+  size?: Size,
+  theme?: any,
+}
+
+class Modal extends React.Component<Props, {}> {
+  constructor(props: Props) {
+    super(props);
+
+    this.handleCloseClick = this.handleCloseClick.bind(this);
+    this.handleToggleClick = this.handleToggleClick.bind(this);
+
+  }
+
+  handleCloseClick(e: any): void {
+    e.preventDefault();
+
+    const props = this.props;
+    const { modal, dialog } = getModalElement(props.id as string);
+
+    props.trigger.animate.out(modal, dialog);
+    setTimeout(() => bodyStyle('', ''), 200);
+  }
+
+  handleToggleClick(e: any): void {
+
+    const props = this.props;
+    const { modal, dialog } = getModalElement(props.id as string);
+
+    const show = (): void => {
+      bodyStyle('16px', 'hidden');
+      props.trigger.animate.in(modal, dialog);
+
+    };
+
+    const hide = (): void => {
```

Comment:
There is no need for `hide()` or `show()` to be arrow functions, please use the standard `if ... else ...` conditionon line 95.